Libraries/LibWeb/HTML/TraversableNavigable.cpp

```diff
@@ -1179,6 +1179,12 @@ TraversableNavigable::HistoryStepResult TraversableNavigable::update_for_navigab
     // 1. Let step be traversable's current session history step.
     auto step = current_session_history_step();
 
+    // AD-HOC: If the parent traversable is inside a cross-document navigation and hasn't updated the target
+    //         step yet, then add 1 to the step.
+    if (m_session_history_step_is_locked) {
+        step++;
+    }
```

Comment:
if reaching `update_for_navigable_creation_or_destruction()` during parent traversable navigation is a problem, then could we rearrange callers of this function to wait for navigation to finish?